### PR TITLE
chore(docs): Add integration docs

### PR DESCRIPTION
**Pull request by boddissattva**
* Menu de gauche
 - Installation
   - local development
   - on-premise
									- Introduction
									- Minimal configuration
										- Maven confguration
										- lockback
										-applicatoin.yml avecle mini
									- detail du reste
										- database
										- logs
										- server (ssl)
										- ....
										
   - advance topic
										- liquibase
										- metrics produite
										- authentification ldap
										- compression
										- session management
										- actuator
										- specifics value
   - CI/CD integration
										- synchronize

### CODE REVIEW

**Review by KarimGl — DISMISSED** [review]

File: docs/integration.md [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+
+!!! important "Chutney rapid technical description"
+    Chutney server is a Spring Boot application running with [Undertow](https://undertow.io/) using existing Spring stack (mvc, webflux, security).
+
+    Chutney UI is an [Angular](https://angular.io/) web application.
+
+    **Chutney follows Angular and Spring Boot last versions and associated dependencies.**
+
+!!! note "Packaging example"
+    The module [local-dev](https://github.com/chutney-testing/chutney/tree/master/packaging/local-dev) is such a packaging.
```
Comment (bessonm) [inline docs/integration.md:14]: Maven module local-dev shows one way of packaging Chutney.
Use it as an example to make your own package.

File: docs/integration.md [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+        <scope>runtime</scope>
+    </dependency>
+    <dependency> <!-- (1)! -->
+        <groupId>com.h2database</groupId>
+        <artifactId>h2</artifactId>
+    </dependency>
+    <dependency> <!-- (2)! -->
+        <groupId>org.postgresql</groupId>
+        <artifactId>postgresql</artifactId>
+    </dependency>
```
Comment (bessonm) [inline docs/integration.md:80]: I would emphasize and do another block about database integration

File: docs/integration.md [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+    ```
+
+    1. Required if using a H2 as Chutney main database.
```
Comment (bessonm) [inline docs/integration.md:83]: weird phrasing...
maybe just "I you want to use H2 as (...)"

**Review by bessonm — CHANGES_REQUESTED** [review]

File: docs/integration.md [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+!!! warning "Handling secrets"
+    We do not discuss here how one could handle secrets present in configuration files because this could vary extensively on the deployment process applied.
+    
+    For example, if [Ansible](https://docs.ansible.com/ansible/latest/index.html) is used, one could package a subset of configuration files to let them be filtered at deployment time and be included in the runtime classpath of the application. 
+
+## Database
+
+Chutney relies internally on a relational database schema embedded and abstracted by [Liquidbase](https://www.liquibase.org/).
+
+Associated changelog could be found [here](https://github.com/chutney-testing/chutney/blob/master/server/src/main/resources/changelog/db.changelog-master.xml).
```
Comment (bessonm) [inline docs/integration.md:130]: You can find corresponding changelog here

File: docs/integration.md [hunk 5]
```diff
@@ -0,0 +1,435 @@
...
+
+The configuration of the associated datasource is done by setting **spring.datasource** properties:
+=== "H2 (memory)"
+    ``` yaml
+    spring:
+        datasource:
+            url: jdbc:h2:mem:dbName
+    ```
+    !!! note
+        The module [local-dev](https://github.com/chutney-testing/chutney/tree/master/packaging/local-dev) is an example of an H2 embedded configuration with filesystem persistence.
```
Comment (bessonm) [inline docs/integration.md:143]: If you want to see an example, maven module local-dev uses an embedded H2 with filesystem persistence.

File: docs/integration.md [hunk 6]
```diff
@@ -0,0 +1,435 @@
...
+## Database
+
+Chutney relies internally on a relational database schema embedded and abstracted by [Liquidbase](https://www.liquibase.org/).
+
+Associated changelog could be found [here](https://github.com/chutney-testing/chutney/blob/master/server/src/main/resources/changelog/db.changelog-master.xml).
+
+!!! note
+    **Chutney has been tested with H2 and PostgreSQL databases.**
+
+The configuration of the associated datasource is done by setting **spring.datasource** properties:
```
Comment (bessonm) [inline docs/integration.md:135]: Use the property **spring.datasource** to configure your datasource

File: docs/integration.md [hunk 7]
```diff
@@ -0,0 +1,435 @@
...
+            username: user
+    ```
+
+## Authentication & Authorization
+
+!!! important
+    The module [local-dev](https://github.com/chutney-testing/chutney/tree/master/packaging/local-dev) presents an example including :
+
+    * Use of the **mem-auth** profile.
+    * An example of a custom LDAP authentication (the LDAP server is an embedded one).
```
Comment (bessonm) [inline docs/integration.md:159]: Maven module [local-dev](https://github.com/chutney-testing/chutney/tree/master/packaging/local-dev) shows :

    * How to use in memory authentication and roles, see the **mem-auth** profile.
    * How to use a custom LDAP authentication (for example purpose, it uses an embedded LDAP server).

**Review by bessonm — CHANGES_REQUESTED** [review]

File: docs/integration.md [hunk 8]
```diff
@@ -0,0 +1,466 @@
...
+
+If you want to add another authentication mechanism, you should follow the [Spring security architecture](https://spring.io/guides/topicals/spring-security-architecture).
+
+!!! important "Authentication requirements"
+    The principal build by the authentication mechanism must be an instance of the Chutney [UserDto](https://github.com/chutney-testing/chutney/blob/master/server/src/main/java/com/chutneytesting/security/api/UserDto.java).
+    // I don't understand what you meant
+
+User roles and permissions are configured either with Web app form or by editing the file.
+
+One could use the existing [AuthenticationService](https://github.com/chutney-testing/chutney/blob/master/server/src/main/java/com/chutneytesting/security/domain/AuthenticationService.java) Chutney Spring Bean to retrieve Chutney roles by user id and grant associated authorities. // I don't understand what you mean, is it useful ? provide a real use case for showing why and how it could be done
```
Comment (bessonm) [inline docs/integration.md:234]: Sorry, I did not understand what was the point here

File: docs/integration.md [hunk 9]
```diff
@@ -0,0 +1,466 @@
...
+    
+    1. Optional, if the role include the characters 'admin', ignoring case, all permissions will be granted to that user.
+
+!!! warning
+    If you create a role name including characters 'admin' (ignoring case), all permissions will be granted to users with this role.
+
+If you want to add another authentication mechanism, you should follow the [Spring security architecture](https://spring.io/guides/topicals/spring-security-architecture).
+
+!!! important "Authentication requirements"
+    The principal build by the authentication mechanism must be an instance of the Chutney [UserDto](https://github.com/chutney-testing/chutney/blob/master/server/src/main/java/com/chutneytesting/security/api/UserDto.java).
```
Comment (bessonm) [inline docs/integration.md:229]: I don't understand what you mean

**Review by bessonm — APPROVED** [review]

File: docs/integration.md (hunk 8 above)
Comment (boddissattva) [reply docs/integration.md:234]: If a new authentication mechanism wants to rely on role-user id Chutney mappings, it must retrieve the role assciated to the user just authenticated and grant all the role's authorities. The service mentionned here has a method which retrieve the Chutney role associated with the user id passed in.

As authorities arae just strings, another strategy for a new authentication mechanism would be to grant authorities after some user attributes retrieved during the authentication.

File: docs/integration.md (hunk 9 above)
Comment (boddissattva) [reply docs/integration.md:229]: Spring Security relies on an interface (UserDetails i think) which represents a user. This one must be implemented to specify what is a user for a specific application. For Chutey, it's the UserDto class.
If an authentiation mechanism create a Userdetails instance of another type, some Chutney functionalities break.

File: docs/installation/local_dev.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
+
+You can run scenarios without installing a Chutney server. This fits well within a CI or for a developer local setup.
+
+However, building software is most often a teamwork !  
+Doing so, you will need to collaborate and share scenarios, track their executions 
+and allow functional and business analyst to review and be involved in testing their product.
+
+That's why we provide a server and web UI to help us do all these things.
+
+You can find all code and configuration below in this [example project](https://github.com/chutney-testing/chutney-project-template)
```
Comment (KarimGl) [inline docs/installation/local_dev.md:10]: open external link in new tab.
Do it for all other external links in this PR

File: docs/installation/on_premise/minimal_conf.md [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+        <artifactId>wlthinclient</artifactId>
+        <version>x.x.x</version>
+        <scope>runtime</scope>
+    </dependency>
+    ```
+
+    1. Example for using [SQL actions](/documentation/actions/sql) and query an Oracle database
+    2. Example for using [JMS actions](/documentation/actions/jms) with a WebLogic server
+
+Finally, add your own [Actions](/documentation/actions/introduction) and [Functions](/documentation/functions/classpath)
```
Comment (KarimGl) [inline docs/installation/on_premise/minimal_conf.md:104]: How to add custom actions and functions is not documented.
May be document it in an other page and link it here ?

File: docs/installation/ci_cd/scenario_sync.md [hunk 12]
```diff
@@ -1,33 +1,3 @@
...
-    1. Download the latest release jar [chutney-local-dev-x.x.x.jar](https://github.com/chutney-testing/chutney/releases/latest).
-    2. Start Chutney locally with `java -jar chutney-local-dev-x.x.x.jar`
-
-!!! note "Intellij"
-
-    1. Checkout [chutney](https://github.com/chutney-testing/chutney).
-    2. Build the project using maven : `mvn compile [-DuseExternalNpm]`
-    3. Start [Intellij run configuration](https://www.jetbrains.com/help/idea/run-debug-configuration.html) `start_local_server`
-
 # Synchronize *your* kotlin scenario with a server
```
Comment (KarimGl) [inline docs/installation/ci_cd/scenario_sync.md:1]: update this doc.
Synchronize function can now create scenarios.

File: docs/installation/on_premise/minimal_conf.md (hunk 11 above)
Comment (bessonm) [reply docs/installation/on_premise/minimal_conf.md]: done

**Review by nbrouand — APPROVED** [review]

**Pull request merged**